Realtime/ClientOptions.cs

```diff
@@ -11,17 +12,17 @@ public class ClientOptions
         /// <summary>
         /// The function to encode outgoing messages. Defaults to JSON
         /// </summary>
-        public Action<object, Action<string>> Encode { get; set; } = (payload, callback) => callback(JsonConvert.SerializeObject(payload,Client.Instance.SerializerSettings));
+        public Action<object, Action<string>>? Encode { get; set; }
 
         /// <summary>
         /// The function to decode incoming messages.
         /// </summary>
-        public Action<string, Action<SocketResponse>> Decode { get; set; } = (payload, callback) => callback(JsonConvert.DeserializeObject<SocketResponse>(payload, Client.Instance.SerializerSettings));
+        public Action<string, Action<SocketResponse?>>? Decode { get; set; }
```

Comment:
The Socket code that uses this asserts that the value is not null with the `!` postfix operator.  I don't see anything obvious that actually verifies this so it might be a good idea to add something or change this to be non-nullable.

Reply:
It's set in the constructor!

https://github.com/supabase-community/realtime-csharp/blob/cca23d27a9a4027a8f5196265eb95ad4d134ea8d/Realtime/Client.cs#L113-L135